contracts/xasset/src/error.rs

```diff
@@ -75,10 +75,19 @@ pub enum Error {
 
     // Payment exceeds interest due
     PaymentExceedsInterestDue = 24,
-    
+
     // "Interest must be paid before debt can be repaid"
     InterestMustBePaidFirst = 25,
 
     // "Insufficient XLM to pay interest"
     InsufficientXLMForInterest = 26,
+
+    // Approval needed for interest repayment
+    InsufficientApprovedXLMForInterestRepayment = 27,
+
+    // Invoking XLM SAC contract failed
+    XLMHostInvocationFailed = 28,
```

Comment:
Not sure the word "Host" is useful here
```suggestion
    XLMInvocationFailed = 28,
```